public/app/features/dashboard-scene/settings/AnnotationsEditView.tsx

```diff
@@ -36,72 +46,221 @@ export class AnnotationsEditView extends SceneObjectBase<AnnotationsEditViewStat
+    layers.splice(this.getAnnotationsLength(), 0, newAnnotation);
+
+    data.setState({
+      layers,
+    });
+
+    this.setState({ editIndex: this.getAnnotationsLength() - 1 });
   };
 
   public onEdit = (idx: number) => {
-    console.log('todo: onEdit');
+    this.setState({ editIndex: idx });
+  };
+
+  public goBackToList = () => {
+    this.setState({ editIndex: undefined });
+  };
+
+  public onPreview = () => {
+    this.setState({ editIndex: undefined });
+
+    this._dashboard.stopUrlSync();
+
+    locationService.partial({ editview: null });
+
+    this._dashboard.startUrlSync();
   };
 
-  public onMove = (idx: number, direction: number) => {
-    console.log('todo: onMove');
+  public onMove = (idx: number, direction: MoveDirection) => {
+    const data = this.getSceneDataLayers();
+
+    const layers = [...data.state.layers];
+    const [layer] = layers.splice(idx, 1);
+    layers.splice(idx + direction, 0, layer);
+
+    data.setState({
+      layers,
+    });
   };
 
   public onDelete = (idx: number) => {
-    console.log('todo: onDelete');
+    const data = this.getSceneDataLayers();
+
+    const layers = [...data.state.layers];
+    layers.splice(idx, 1);
+
+    data.setState({
+      layers,
+    });
+  };
+
+  public onUpdate = (annotation: AnnotationQuery, editIndex: number) => {
+    const data = this.getSceneDataLayers();
+
+    const layers = [...data.state.layers];
+    const layer = layers[editIndex];
+
+    if (layer instanceof dataLayers.AnnotationsDataLayer) {
+      layer.setState({
+        key: `annotations-${annotation.name}`,
+        name: annotation.name,
+        isEnabled: Boolean(annotation.enable),
+        isHidden: Boolean(annotation.hide),
+        query: annotation,
+      });
+
+      //need to rerun the layer to update the query and
+      //see the annotation on the panel
+      layer.runLayer();
+    }
   };
 }
 
 function AnnotationsSettingsView({ model }: SceneComponentProps<AnnotationsEditView>) {
   const dashboard = model.getDashboard();
-  const sceneDataLayers = model.getSceneDataLayers();
+  const { overlay } = dashboard.useState();
+  const { layers } = model.getSceneDataLayers().useState();
   const { navModel, pageNav } = useDashboardEditPageNav(dashboard, model.getUrlKey());
   const { editIndex } = model.useState();
+  const panels = dashboardSceneGraph.getVizPanels(dashboard);
 
-  let annotations: AnnotationQuery[] = [];
+  const annotations: AnnotationQuery[] = dataLayersToAnnotations(layers);
 
-  if (sceneDataLayers) {
-    const { layers } = sceneDataLayers.useState();
-    annotations = dataLayersToAnnotations(layers);
+  if (editIndex != null && editIndex < model.getAnnotationsLength()) {
+    return (
+      <AnnotationsSettingsEditView
+        annotationLayer={model.getDataLayer(editIndex)}
+        pageNav={pageNav}
+        panels={panels}
+        editIndex={editIndex}
+        navModel={navModel}
+        dashboard={dashboard}
+        overlay={overlay}
```

Comment:
Torkel opened this PR #81852 which moves the overlay out of the edit views, so we won't need to add it in the edit views anymore 🙌

Reply:
Removed the overlays